src/probnum/filtsmooth/statespace/discrete_transition_utils.py

```diff
@@ -0,0 +1,50 @@
+import typing
+
+import numpy as np
+
+import probnum.random_variables as pnrv
+
```

Comment:
Good question indeed. I am wondering the same. However, when in doubt I rather make these functions public unless it bites me. Any thoughts about this? (@schmidtjonathan?)

Reply:
Mh, I personally would leave them public since it is a `utils` module, for which the natural intended use is that it is accessed from outside. If the method is expected to be called only in a narrow context, I think it is misplaced in a `utils` file. However, this is probably not the case, since conditioning on measurements is a quite general concept, right?

(My suggestion: leave as is)